File: egghead.io/data-structures-and-algorithms-in-js/lesson-04-stack-data-structure-in-js/custom_stack.py
class CustomStack:
  def __init__(self):
    self.stack = []

  def push(self, item):
    self.stack.append(item)

  def pop(self):
    self.stack.pop()

  def peek(self):
    return self.stack[self.size - 1]

  def size(self):
    return len(self.stack)

  def is_empty(self):
    return self.size() == 0

File: egghead.io/data-structures-and-algorithms-in-js/lesson-04-stack-data-structure-in-js/custom-stack.js
class CustomStack {
  constructor() {
    this.stack = [];
  }

  push(item) {
    this.stack.push(item);
  }

  pop() {
    this.stack.pop();
  }

  peek() {
    return this.stack[this.size - 1];
  }

  get size() {
    return this.stack.length;
  }

  isEmpty() {
    return this.size === 0;
  }
}

module.exports = CustomStack;
